transport/src/test/java/io/camunda/zeebe/transport/stream/impl/StreamIntegrationTest.java

```diff
@@ -101,7 +103,10 @@ private RemoteStreamer<TestSerializableData, TestSerializableData> startRemoteSt
                   remoteStreamService =
                       new TransportFactory(actorScheduler)
                           .createRemoteStreamServer(
-                              serverService, TestSerializableData::new, RemoteStreamMetrics.noop());
+                              serverService,
+                              TestSerializableData::new,
+                              (e, d) -> errorHandler.handleError(e, d),
```

Comment:
This is on purpose in order to handle changes to the `errorHandler` reference. If I pass it here, then it will point to the original errorHandler - changing the value of `errorHandler` here will not be reflected downstream.

I get that it's not obvious - do you have an idea of how we could make it clearer that it's meant to be dynamic to allow us to change the error handler? I could make an actual error handler class to reflect this I guess

```java
public final class ProxiedErrorHandler<P> implements ErrorHandler<P> {
  @Override
  public void handleError(final Throwable error, final P payload) {
    errorHandler.handleError(error, payload);
  }
}
```

Not sure if that makes it clearer though :smile:

Reply:
To keep it simple, may be just add a comment in there, and rename `errorHandler` to something like `runTimeErrorHandler` to indicate it can be overwritten.